### PR TITLE
Implement test cases

### PR DESCRIPTION
**Pull request by killVung**
Sorry about the white spaces problem, I am using Sublime without any plugins, so I don't realized the existence of them.

Also I had fixed some of the problems @hugovk  mentioned

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -7,48 +7,28 @@ class testNormalize(unittest.TestCase):
     """
     A test class for the normalize method.
     """
-    def testFakePackage(self):
+
+    def test_FakePackage(self):
```
Comment (hugovk) [inline tests.py]: -> test_fake_package

File: tests.py [hunk 2]
```diff
@@ -7,48 +7,28 @@ class testNormalize(unittest.TestCase):
...
-                          "Never Gonna Let you Down")
-        self.assertRaises(ValueError,
-                          vanity.normalize,
-                          "Never Gonna Tell a lie")
-        self.assertRaises(ValueError,
-                          vanity.normalize,
-                          "And...")
 
-    def testDjango(self):
+    def test_Django(self):
```
Comment (hugovk) [inline tests.py]: -> test_django

File: tests.py [hunk 3]
```diff
@@ -7,48 +7,28 @@ class testNormalize(unittest.TestCase):
...
         normalized = vanity.normalize("dJaNgO")
         self.assertEqual(normalized, "Django")
 
-    def testNone(self):
+    def test_None(self):
```
Comment (hugovk) [inline tests.py]: -> test_none

File: tests.py [hunk 4]
```diff
@@ -7,48 +7,28 @@ class testNormalize(unittest.TestCase):
...
         normalized = vanity.normalize(None)
         self.assertEqual(normalized, "none")
 
-    def testFlask(self):
+    def test_Flask(self):
```
Comment (hugovk) [inline tests.py]: -> test_flask

File: tests.py [hunk 5]
```diff
@@ -7,48 +7,28 @@ class testNormalize(unittest.TestCase):
...
         normalized = vanity.normalize("fLaSk")
         self.assertEqual(normalized, "Flask")
 
-    def testSpaceString(self):
+    def test_SpaceString(self):
```
Comment (hugovk) [inline tests.py]: -> test_space_string

File: tests.py [hunk 6]
```diff
@@ -8,27 +8,38 @@ class testNormalize(unittest.TestCase):
...
-        self.assertEqual(normalized, "")
+    def test_Flask(self):
+        normalized = vanity.normalize("fLaSk")
+        self.assertEqual(normalized, "Flask")
+
+    def test_SpaceString(self):
+        normalized = vanity.normalize("               Flask                ")
+        self.assertEqual(normalized, "               Flask                ")
+
+    # def testEmpty(self):
```
Comment (hugovk) [inline tests.py]: ->test_empty

File: tests.py [hunk 7]
```diff
@@ -8,27 +8,38 @@ class testNormalize(unittest.TestCase):
...
+    #     """
+    #     TODO: this test is rather slow to run,
+    #     perhaps normalize could be refactored to check this
+    #     and kick out faster.
+    #     """
+    #     normalized = vanity.normalize("")
+    #     self.assertEqual(normalized, "")
```
Comment (hugovk) [inline tests.py]: Why is this test commented out?

File: tests.py [hunk 8]
```diff
@@ -8,27 +8,38 @@ class testNormalize(unittest.TestCase):
     A test class for the normalize method.
     """
 
-    def testFakePackage(self):
+    def test_FakePackage(self):
```
Comment (hugovk) [inline tests.py]: -> test_fake_package

File: tests.py [hunk 9]
```diff
@@ -8,27 +8,38 @@ class testNormalize(unittest.TestCase):
...
         self.assertRaises(ValueError,
                           vanity.normalize,
                           "FAKEPACKAGE1@!")
+        self.assertRaises(ValueError,
+                          vanity.normalize,
+                          "1337INoscopeyou")
 
-    def testDjango(self):
+    def test_Django(self):
```
Comment (hugovk) [inline tests.py]: -> test_django

File: tests.py [hunk 10]
```diff
@@ -8,19 +8,30 @@ class testNormalize(unittest.TestCase):
...
         normalized = vanity.normalize("dJaNgO")
         self.assertEqual(normalized, "Django")
 
-    def testNone(self):
+    def test_None(self):
         normalized = vanity.normalize(None)
         self.assertEqual(normalized, "none")
 
+    def test_Flask(self):
```
Comment (hugovk) [inline tests.py]: -> test_flask

File: tests.py [hunk 11]
```diff
@@ -8,19 +8,30 @@ class testNormalize(unittest.TestCase):
...
+        normalized = vanity.normalize("fLaSk")
+        self.assertEqual(normalized, "Flask")
+
+    def test_SpaceString(self):
```
Comment (hugovk) [inline tests.py]: -> test_space_string

File: tests.py [hunk 12]
```diff
@@ -8,19 +8,30 @@ class testNormalize(unittest.TestCase):
...
+        normalized = vanity.normalize("               Flask                ")
+        self.assertEqual(normalized, "               Flask                ")
+
     def testEmpty(self):
```
Comment (hugovk) [inline tests.py]: -> test_empty

File: tests.py [hunk 13]
```diff
@@ -8,20 +8,31 @@ class testNormalize(unittest.TestCase):
...
-    def testEmpty(self):
+    def test_Flask(self):
+        normalized = vanity.normalize("fLaSk")
+        self.assertEqual(normalized, "Flask")
+
+    def test_SpaceString(self):
+        normalized = vanity.normalize("               Flask                ")
+        self.assertEqual(normalized, "               Flask                ")
+
+    def test_Empty(self):
```
Comment (hugovk) [inline tests.py]: -> test_empty

File: tests.py [hunk 14]
```diff
@@ -8,20 +8,31 @@ class testNormalize(unittest.TestCase):
...
         normalized = vanity.normalize("dJaNgO")
         self.assertEqual(normalized, "Django")
 
-    def testNone(self):
+    def test_None(self):
         normalized = vanity.normalize(None)
         self.assertEqual(normalized, "none")
 
-    def testEmpty(self):
+    def test_Flask(self):
```
Comment (hugovk) [inline tests.py]: -> test_flask

**Comment by killVung** [comment]
Should I also rename the class name to test_Normalize for consistency?

**Comment by hugovk** [comment]
@killVung Yes please.

**Comment by killVung** [comment]
I actually changed it while asking for it 
Have a look

**Comment by hugovk** [comment]
And please lowercase it for consistency too. Thank you!

**Comment by killVung** [comment]
Done

**Comment by hugovk** [comment]
I squashed this into a single commit and merged.

Thanks!

**Comment by killVung** [comment]
Thank you so much
I appreciate it

**Pull request merged**